### ISSUE TITLE
Facebook provider doesn't work with Facebook API version 'v2.3' and above

### ISSUE THREAD

**Issue by Folleo** [Sorcery/sorcery#52]
Facebook changed `access_token` format response in version 2.3, so now it returns `:json` format. See https://developers.facebook.com/docs/apps/changelog

Oauth2 gem can handle `:json`, but in facebook provider there is no option to change `:parse` option from `:query` to `:json`.

In `facebook.rb` in method `process_callback` we have `:parse` option:
```
def process_callback(params, _session)
        args = {}.tap do |a|
          a[:code] = params[:code] if params[:code]
        end

        get_access_token(args, token_url: token_url, mode: mode,
                               param_name: param_name, parse: parse)
end
```

But it's always `:query` (`@parse = :query`) and there is no ability to change it via config.

**Comment by jeriko** [comment]
What worked for me was `config.facebook.instance_variable_set "@parse", :json` in the sorcery initializer. Bit hacky but my users can login via Facebook again.

**Comment by Folleo** [comment]
@jeriko thanks man, it helped

**Comment by milanij** [comment]
@jeriko Wow! Thanks! This really helped!

**Comment by athix** [comment]
Fixed by Sorcery 0.11.0
